rs/tests/src/qualification/steps/workload.rs

```diff
@@ -0,0 +1,29 @@
+use super::Step;
+
+#[derive(Clone)]
+pub struct Workload {
+    pub message_size: usize,
+    pub rps: f64,
+}
+
+impl Step for Workload {
+    fn execute(
+        &self,
+        env: ic_system_test_driver::driver::test_env::TestEnv,
+        rt: tokio::runtime::Handle,
+    ) -> anyhow::Result<()> {
+        // Small messages
+        ic_consensus_system_test_utils::performance::test_with_rt_handle(
+            env,
+            self.message_size,
+            self.rps,
+            rt,
+            false,
+        )
+        .map(|_| ())
+    }
+
+    fn max_retries(&self) -> usize {
+        3
```

Comment:
It seems like this is the only step with retries, why is that case?

Reply:
The test itself (including with the spawning of a new testnet) is retried 3 times. I wanted to separate some flaky parts into its own retry logic just so we have quicker retries of flaky parts. Only workload and xnet tests should be retried for more than 1 because other steps result in an immediate failure if they are not successful. If the upgrade of a subnet fails, I doubt that one more try for it will result in it succeeding right?

Reply:
I guess I was more surprised to the test being retried at all. As far as I can tell the original consensus performance test [isn't marked as flaky](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/tests/consensus/BUILD.bazel?L347). Or is the problem that consensus performance test is usually ran on the performance testnet?

Reply:
That are all the right questions. Our previous implementation of qualification had retries for all steps. When doing this migration I tried to bring some sense into what can and can't be retried. We use default nodes and I thought that resources could be the limit so that is why put a retry there just to be sure. If you think its not needed I will gladly remove it

Reply:
The goal is to detect performance deg. I am all for putting some sensible parameters which are defined [here](https://github.com/dfinity/ic/pull/1227/files#diff-6cd65738e6bfa607803d687714c755e8f59b2965ee0fbd72d223fb73cde8c668R138). The topology we use is 2 app subnets and 1 system subnets with 4 unassigned nodes. All of the nodes are using default resources. I would just like to say that even if we tweak the parameters and they succeed on the first try all the time I would still like to keep this mechanism for the transition period so our team is familiar with the flow of quali. When we are sure this is what we want to use, we will move the mechanism away

Reply:
Sure, that makes sense! We can keep it until we find a configuration that we are comfortable with

Reply:
Decided to take it out because both of those tests are not considered flaky 🥇. Thanks for swift responses!